pkg/users/group-actions.jsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+
+import { Dropdown, DropdownItem, KebabToggle } from '@patternfly/react-core';
+
+import { delete_group_dialog } from "./delete-group-dialog.js";
+import { rename_group_dialog } from "./rename-group-dialog.js";
+
+const _ = cockpit.gettext;
+
+export const GroupActions = ({ group, accounts, isMainPage }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
+
+    const actionsMain = [
+        <DropdownItem key="edit-group"
+                      onClick={ev => { ev.preventDefault(); cockpit.location.go(["group", group.name]) }}>
+            {isMainPage ? _("Edit group") : _("Edit")}
+        </DropdownItem>,
+    ];
+    const actions = [
+        <DropdownItem key="rename-group"
+                      onClick={() => { setKebabOpen(false); rename_group_dialog(group.name) }}>
+            {isMainPage ? _("Rename group") : _("Rename")}
```

Comment:
I kinda agree with @martinpitt that 'rename' is actually editing and can be ommited as a standalone operation in the main page per-group action dropdown.

Reply:
Renaming is sometimes possibly problematic, as it would need to change the associations with all users, could impact other things such as the `sudoers` file (and `/etc/sudoers.d/`), and files may have permissions with the group; therefore it should probably only be exposed in the group details page.

We should also do what we can (like change user association and _possibly_ sudoers — or at least warn of `sudoers` if we can) and warn about the rest (changing the group ownership of files, for example). We could probably change the ownership of files too, but then there's the issue of multiple filesystems (including remote ones) and unmounted storage (USB disks or unmounted network storage).

Since we can't handle every case, we should warn about it. Between it being "risky" and also a not so common modification, we should not have it be an option in the top-level, meaning that we should only expose it from the group details page.

This also means we would want to improve the rename modal with related warnings and make it a warn button state instead of primary.

Reply:
Note that file ownership is not affected by renaming a group -- the file system stores user/group IDs only, and the IDs cannot be edited. But you are right that group *names* are being used in places like /etc/sudoers, so e.g. renaming the `wheel` group could easily lock you out from your own system.

I'm not even sure if it's a good idea to expose group renaming as a feature at all, but if we do, it should be a bit more obscured indeed.